src/lib/data.ts

```diff
@@ -194,8 +194,21 @@ function consolidate_implementation_reports(implementations: ImplementationRepor
             // Minor side effect trick: for each key there is at least one Test Result that has a value. We can therefore
             // safely set the result to 'false' temporarily if the test result is not available; there must be a false or true
             // value somewhere else in the array anyway.
-            const all_results: boolean[] = variant_results.map((results: TestResults): boolean => key in results ? results[key] : false);
-            retval[key] = all_results.reduce( (p: boolean, c: boolean): boolean => p || c, false);
+            const all_results: (boolean|string)[] = variant_results.map((results: TestResults): (boolean|string) => { 
```

Comment:
These types are getting a little ugly. If you wanted to you could add a map step that converts the Boolean|string to an Enum with Untested, Pass, Fail, NA values. But entirely optional, fine as is.

Reply:
I agree it is something to be done. At this moment, I would add it to a TODO list for after merging this PR; it is important to introduce the functionality into the system and then do this change behind the scenes, so to say.